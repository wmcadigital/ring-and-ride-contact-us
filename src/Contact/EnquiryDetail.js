import { useFormState } from "react-final-form";

import EnquiryTypes from "./EnquiryTypes";
import { required } from "../common/validation";
import FormSection from "../common/FormSection";
import TextAreaContainer from "../common/TextAreaContainer";
import ProgressIndicator from "../common/ProgressIndicator";
import Question from "../common/Question";
import TextArea from "../common/TextArea";

const EnquiryDetail = () => {
  const stateApi = useFormState();
  const formValues = stateApi.values;
  const enquiryLabel = EnquiryTypes[formValues.enquiryType]?.detailLabel;

  const error = stateApi.submitFailed ? stateApi.errors?.enquiryDetail : null;

  return (
    <FormSection>
      <ProgressIndicator
        sectionPosition="Section 1 of 2"
        sectionName="About your enquiry"
      />
      <TextAreaContainer>
        <Question text={`What is your ${enquiryLabel} enquiry about?`} />
        <TextArea
          fieldName="enquiryDetail"
          validation={required}
          error={error}
        />
      </TextAreaContainer>
    </FormSection>
  );
};

export default EnquiryDetail;
