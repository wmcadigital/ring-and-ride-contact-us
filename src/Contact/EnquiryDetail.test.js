import { useFormState } from "react-final-form";
import { createRenderer } from "react-test-renderer/shallow";

import EnquiryDetail from "./EnquiryDetail";

jest.mock("react-final-form");

describe("EnquiryDetail", () => {
  it("renders as expected", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        enquiryType: "lostProperty",
        enquiryDetail: "test enquiry details",
      },
    }));
    const renderer = createRenderer();

    renderer.render(<EnquiryDetail />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("renders with an error", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        enquiryType: "lostProperty",
      },
      submitFailed: true,
      errors: { enquiryDetail: "Required" },
    }));
    const renderer = createRenderer();

    renderer.render(<EnquiryDetail />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
