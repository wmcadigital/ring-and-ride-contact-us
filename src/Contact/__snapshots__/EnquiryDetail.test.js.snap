// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnquiryDetail renders as expected 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="About your enquiry"
    sectionPosition="Section 1 of 2"
  />
  <TextAreaContainer>
    <Question
      text="What is your lost property enquiry about?"
    />
    <TextArea
      error={null}
      fieldName="enquiryDetail"
      validation={[Function]}
    />
  </TextAreaContainer>
</FormSection>
`;

exports[`EnquiryDetail renders with an error 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="About your enquiry"
    sectionPosition="Section 1 of 2"
  />
  <TextAreaContainer>
    <Question
      text="What is your lost property enquiry about?"
    />
    <TextArea
      error="Required"
      fieldName="enquiryDetail"
      validation={[Function]}
    />
  </TextAreaContainer>
</FormSection>
`;
